### PR TITLE
Catch an exception in get-all

### PR DESCRIPTION
**Pull request by yito88**
## Description
In `transfer` and `transfer-append` tests in ScalarDB tests, `get-all` which reads all the final state misbehaved.
The dummy writes were committed even though the read failed due to conflicts. Then the `get-all` transaction was retried.
It caused the unexpected version increasing. That's why the check phase failed.

## Related issues and/or PRs
Daily Jepsen test failed

## Changes made
- Catch the exceptions when transaction reads

## Checklist

> The following is a best-effort checklist. If any items in this checklist are not applicable to this PR or are dependent on other, unmerged PRs, please still mark the checkboxes after you have read and understood each item.

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

> Provide any additional information or notes that may be relevant to the reviewers or stakeholders.

### CODE REVIEW

File: scalardb/test/scalardb/core_test.clj [hunk 1]
```diff
@@ -25,7 +25,9 @@
       (condp = column
         "tx_id" (Optional/of (TextValue. column id))
         "tx_created_at" (Optional/of (BigIntValue. column (long 1566376246)))
-        "tx_state" (Optional/of (IntValue. column (Integer/parseInt id)))))))
+        "tx_state" (Optional/of (IntValue. column (Integer/parseInt id)))
+        ;; for the coordinator table
```
Comment (yito88) [inline scalardb/test/scalardb/core_test.clj:29]: This is unrelated to the bugfix.
For The group commit feature in the latest ScalarDB.

**Review by inv-jishnu — APPROVED** [review]
LGTM!
Thank you.

**Review by brfrn169 — APPROVED** [review]
LGTM! Thank you!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
